File: frontend/src/types/clip.ts
export interface Clip {
  id: string
  job_id: string
  clip_index: number
  score: number
  duration: number
  transcript: string
  has_face: boolean
  emotion: string | null
  primary_type: string
  tags: string[]
  video_url: string
  thumbnail_url: string
  source_title: string
  start_time: number
  end_time: number
}

File: backend/app/models/clip.py
from __future__ import annotations

from pydantic import BaseModel


class ClipResponse(BaseModel):
    id: str
    job_id: str
    clip_index: int
    score: float
    duration: float
    transcript: str
    has_face: bool
    emotion: str | None = None
    primary_type: str
    tags: list[str] = []
    video_url: str
    thumbnail_url: str
    source_title: str
    start_time: float
    end_time: float
